lib/services/upload_service.rb

```diff
@@ -3,22 +3,24 @@
 
 module Services
     class UploadService
-        def initialize()
-            @client = make_s3_client
+        attr_accessor :client
+
+        def initialize(client = make_s3_client)
+            @client = client
+            @filename = SecureRandom.hex(10) # for obfuscation purposes
         end
 
-        def get_download_url(filename)
-            "https://#{ENV['AWS_S3_BUCKET']}.s3.#{ENV['AWS_S3_REGION']}.amazonaws.com/#{filename}"
+        def get_download_url()
```

Comment:
It's not idiomatic ruby to put the prefix `get_` in front of getters. It's also not idiomatic to use parens if there are no arguments needed